src/TypeScript/Rules/RuleDefinitionsProvider.cs

```diff
@@ -0,0 +1,80 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Linq;
+using System.Text;
+using Newtonsoft.Json;
+
+namespace SonarLint.VisualStudio.TypeScript.Rules
+{
+    internal interface ITypeScriptRuleDefinitionsProvider
+    {
+        /// <summary>
+        /// Returns the metadata descriptions for all TypeScript rules
+        /// </summary>
+        IEnumerable<RuleDefinition> GetDefinitions();
+    }
+
+    internal interface IJavaScriptRuleDefinitionsProvider
+    {
+        /// <summary>
+        /// Returns the metadata descriptions for all JavaScript rules
+        /// </summary>
+        IEnumerable<RuleDefinition> GetDefinitions();
+    }
```

Comment:
Separate interfaces for TS and JS.
It's not possible for those interfaces to inherit from a common interface; the compiler complains, even if you try to implement the interfaces explicitly.

In fact, if the interfaces do inherit from a common base you can't explicitly implement `ITypeScriptXXX.GetX` and `IJavaScriptXXX.GetX` separately - you have to implement `IBaseInterface.GetX` i.e. both methods would have the same implementation, which isn't what we want.